src/targets/node/nodeAttacher.ts

```diff
@@ -46,5 +56,100 @@ export class NodeAttacher extends NodeLauncherBase<INodeAttachConfiguration> {
         this.onProgramTerminated(result);
       }
     });
+
+    this.capturedBreaker = false;
+  }
+
+  /**
+   * @inheritdoc
+   */
+  protected createLifecycle(
+    cdp: Cdp.Api,
+    run: IRunData<INodeAttachConfiguration>,
+  ): INodeTargetLifecycleHooks {
+    if (this.capturedBreaker) {
+      return {};
+    }
+
+    // We use a lease file to indicate to the process that the debugger is
+    // still running. This is needed because once we attach, we set the
+    // NODE_OPTIONS for the process, forever. We can try to unset this on
+    // close, but this isn't reliable as it's always possible
+    const leaseFile = new LeaseFile();
+
+    let hitFirstBreakpoint = false;
+    this.capturedBreaker = true;
+
+    return {
+      initialized: async () => {
+        await cdp.Debugger.pause({});
+      },
+      paused: async (_target, ev) => {
+        if (hitFirstBreakpoint) {
+          return false;
+        }
+
+        hitFirstBreakpoint = true;
+        await Promise.all([
+          this.gatherTelemetry(cdp),
+          this.setEnvironmentVariables(cdp, run, leaseFile.path),
+        ]);
+        await cdp.Debugger.resume({});
+
+        return true;
+      },
+      close: async () => {
+        leaseFile.dispose();
+      },
+    };
+  }
+
+  private async setEnvironmentVariables(
+    cdp: Cdp.Api,
+    run: IRunData<INodeAttachConfiguration>,
+    leasePath: string,
+  ) {
+    if (!run.params.attachSpawnedProcesses) {
+      return;
+    }
+
+    if (!isLoopback(run.params.address)) {
```

Comment:
(Because we need to --require the bootstrap file, which almost certainly won't exist on remote systems. Maybe one toggleable approach would be to send its contents over CDP and write it to a tmpdir on the remote machine.)